### ISSUE TITLE
`noInitialRun` not respected by Safari

### ISSUE THREAD

**Issue by tomayac** [emscripten-core/emscripten#19622]
**Version of emscripten/emsdk:**

```bash
$ emcc --version
emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 3.1.36-git
Copyright (C) 2014 the Emscripten authors (see AUTHORS.txt)
This is free and open source software under the MIT license.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**Failing command line in full:**

```bash
# Download https://potrace.sourceforge.net/download/1.16/potrace-1.16.tar.gz.
# Extract it and `cd` into the `potrace-1.16/` folder.
# 
$ emconfigure ./configure --host=wasm32 CFLAGS='-sFILESYSTEM=1 -sEXPORTED_RUNTIME_METHODS=FS,callMain -sMODULARIZE=1 -sEXPORT_ES6'
$ emmake make
```

**Bug report:**

1. Open https://mirage-diligent-partner.glitch.me/ in Chrome. Nothing happens.
   ![Screenshot 2023-06-14 at 12 48 03](https://github.com/emscripten-core/emscripten/assets/145676/755f9103-cfcc-47ae-bd21-952125a7fcde)
1. Open https://mirage-diligent-partner.glitch.me/ in Safari. The code executes.
   ![Screenshot 2023-06-14 at 12 48 55](https://github.com/emscripten-core/emscripten/assets/145676/3e6bc535-aae5-4948-b74e-b60e963834d3)

This is the JavaScript code in question:

```js
import loadWASM from './mkbitmap.js';

const run = async () => {
 const Module = await loadWASM({
   noInitialRun: true,
 }); 
};

run();
```

**Comment by kleisauke** [comment]
This is probably a duplicate of #18357, which was erroneously reintroduced with PR #18551. 

The underlying WebKit issue is tracked at https://bugs.webkit.org/show_bug.cgi?id=223533. A possible workaround is to also build with `-sENVIRONMENT=web`, which would avoid the emit of this `async` function.
https://github.com/emscripten-core/emscripten/blob/a60507f49b83c12b608bafb82968fb3ae9e54b3c/emcc.py#L3854-L3859

**Comment by sbc100** [comment]
Closing as duplicate of #18357.  I'm working on a fix, hopefully I can use jsc to verify the fix on the command line on linux (See #19626)
